### PR TITLE
Improved docs for fulltext search, added psql option to crate.yml

### PR DESCRIPTION
**Pull request by celaus**

### CODE REVIEW

File: blackbox/docs/sql/fulltext.txt [hunk 1]
```diff
@@ -344,13 +338,12 @@ Anyway let's do it here for demonstration purpose::
     +-----------+------------+
     SELECT 2 rows in set (... sec)
 
-As you maybe noticed, the :ref:`_score <sql_ddl_system_column_score>` value has changed for the
-same query text and document because it's a ratio related to all results,
-and if we limit it by filtering on :ref:`_score <sql_ddl_system_column_score>`,
-we'll have fewer or different results.
+As you may have noticed, the :ref:`_score <sql_ddl_system_column_score>` value has changed for the
```
Comment (therealpadams) [inline blackbox/docs/sql/fulltext.txt]: s/might/may

File: app/src/main/dist/config/crate.yml [hunk 2]
```diff
@@ -858,3 +858,13 @@
...
+
+################### POSTGRES WIRE PROTOCOL SUPPORT ###################
+
+# Crate contains experimental support for the PostgreSQL wire protocol v3
+# and emulates a PostgreSQL server v9.5. For further information read our
+# documentation: https://crate.io/docs/reference/en/latest/protocols/postgres.html
+#
+# To connect to Crate via Postgres, use port 5432.
+
+#psql.enabled: false
```
Comment (matriv) [inline app/src/main/dist/config/crate.yml]: I would add the psql.port

File: app/src/main/dist/config/crate.yml [hunk 3]
```diff
@@ -858,3 +858,13 @@
...
 #
 #repositories.url.allowed_urls: ["http://example.org/root/*", "https://*.mydomain.com/*?*#*"]
+
+################### POSTGRES WIRE PROTOCOL SUPPORT ###################
+
+# Crate contains experimental support for the PostgreSQL wire protocol v3
+# and emulates a PostgreSQL server v9.5. For further information read our
+# documentation: https://crate.io/docs/reference/en/latest/protocols/postgres.html
+#
+# To connect to Crate via Postgres, use port 5432.
```
Comment (matriv) [inline app/src/main/dist/config/crate.yml]: Properly adjust according to comment that will be used for psql.port

File: blackbox/docs/sql/fulltext.txt [hunk 4]
```diff
@@ -5,10 +5,10 @@
 Fulltext Search
 ===============
 
-In order to use fulltext searches on columns, one must create a
-:ref:`fulltext index with an analyzer <sql_ddl_index_fulltext>` on the related column
-upfront on table creation, see :ref:`fulltext-indices` for details.
-
+In order to use fulltext search on one or more columns, a
+:ref:`fulltext index with an analyzer <sql_ddl_index_fulltext>` has to be
+defined while creating the table. For more information
```
Comment (matriv) [inline blackbox/docs/sql/fulltext.txt]: I would say For more information, or For more details and not use both in the same phrase.

Comment (matriv) [inline blackbox/docs/sql/fulltext.txt]: I would rephrase to has to be defined on the column(s) when adding them to the table during its creation or altering (or alternation?)

File: blackbox/docs/sql/fulltext.txt [hunk 5]
```diff
@@ -26,26 +26,25 @@ Synopsis
...
+it's not possible to query multiple index columns with different index types
+within the same MATCH predicate.
 
-To get the relevance of a matching row, an internal system column
-:ref:`_score <sql_ddl_system_column_score>` can be selected.
-It is a numeric value which is not absolute but relative to the other rows.
-The higher the score value, the more relevant the row::
+To get the relevance of a matching row, a specific system column
+:ref:`_score <sql_ddl_system_column_score>` can be selected. It contains a
+numeric score *relative to the other rows*: The higher, the more relevant the row::
```
Comment (matriv) [inline blackbox/docs/sql/fulltext.txt]: The higher the score, the more ...

File: blackbox/docs/sql/fulltext.txt [hunk 6]
```diff
@@ -103,28 +101,28 @@ Match Types
...
-               All tokens have to be present in at least one column.
-               So querying for ``foo bar`` should have the tokens ``foo`` in one column and ``bar`` in the same or any other.
+:cross_fields: This match type analyzes the ``query_term`` into tokens and searches all tokens
+               in all given columns at once as if they were one big column (given they have the same analyzer).
+               All tokens have to be present in at least one column, so querying for ``foo bar``
+               should have the tokens ``foo`` in one column and ``bar`` in the same or any other.
 
 :phrase: This match type differs from ``best_fields`` in that it constructs a phrase query from the ``query_term``.
-         A phrase query will only match if the tokens in the columns are in the same order than the
+         A phrase query will only match if the tokens in the columns are *exactly* in the same order than the
```
Comment (matriv) [inline blackbox/docs/sql/fulltext.txt]: in the same order AS the... (as instead of than)

File: blackbox/docs/sql/fulltext.txt [hunk 7]
```diff
@@ -251,16 +244,16 @@ There are two possibilities if a search should span the contents of multiple col
...
 
-When querying multiple columns, there are many ways how the resulting match relevance a.k.a.
-:ref:`_score <sql_ddl_system_column_score>` can be computed, that means which documents match and
-in which order. These different techniques are called :ref:`predicates_match_types`.
+When querying multiple columns, there are many ways how the relevance a.k.a.
+:ref:`_score <sql_ddl_system_column_score>` can be computed. These different
+techniques are called :ref:`predicates_match_types`.
 
-To increase the relevance of rows where one columns matches extremely
+To increase the relevance of rows where one columns match extremely
```
Comment (matriv) [inline blackbox/docs/sql/fulltext.txt]: column not columns

File: blackbox/docs/sql/fulltext.txt [hunk 8]
```diff
@@ -344,13 +338,12 @@ Anyway let's do it here for demonstration purpose::
...
+and by filtering on :ref:`_score <sql_ddl_system_column_score>`,
+'all results' has already changed.
 
 .. warning::
 
-  As noted above :ref:`_score <sql_ddl_system_column_score>` is a ratio and not comparable
-  across search so use it only for filtering if you know what you're doing and on your own risk.
-
+  As noted above :ref:`_score <sql_ddl_system_column_score>` is a relative number and not comparable
+  across searches. The use for filtering is greatly discouraged.
```
Comment (matriv) [inline blackbox/docs/sql/fulltext.txt]: The usage

File: blackbox/docs/sql/fulltext.txt [hunk 9]
```diff
@@ -344,13 +338,12 @@ Anyway let's do it here for demonstration purpose::
     +-----------+------------+
     SELECT 2 rows in set (... sec)
 
-As you maybe noticed, the :ref:`_score <sql_ddl_system_column_score>` value has changed for the
-same query text and document because it's a ratio related to all results,
-and if we limit it by filtering on :ref:`_score <sql_ddl_system_column_score>`,
-we'll have fewer or different results.
+As you might have noticed, the :ref:`_score <sql_ddl_system_column_score>` value has changed for the
```
Comment (matriv) [inline blackbox/docs/sql/fulltext.txt]: may have noticed

Comment (celaus) [inline blackbox/docs/sql/fulltext.txt]: no
paul said might

File: blackbox/docs/sql/fulltext.txt (hunk 5 above)
Comment (celaus) [inline blackbox/docs/sql/fulltext.txt]: Context should tell you that we are talking about the score.

File: app/src/main/dist/config/crate.yml (hunk 2 above)
Comment (celaus) [inline app/src/main/dist/config/crate.yml]: there is a psql.port?! :D

**Comment by matriv** [comment]
Igtm

File: app/src/main/dist/config/crate.yml [hunk 10]
```diff
@@ -858,3 +858,14 @@
 # "file" urls must be prefixed with an entry configured in ``path.repo``
 #
 #repositories.url.allowed_urls: ["http://example.org/root/*", "https://*.mydomain.com/*?*#*"]
+
+################### POSTGRES WIRE PROTOCOL SUPPORT ###################
```
Comment (jodok) [inline app/src/main/dist/config/crate.yml]: i'd rather not use all capital letters. most of the section are camelcase
